Taskfile.yml

```diff
@@ -36,8 +25,43 @@ tasks:
       - go mod tidy
       - golangci-lint run --fix
 
+  setup:
+    desc: Setup linter, formatter, etc. for local testing and CI
+    cmds:
+      - task: install-gofumpt
+      - task: install-golangci
+
   test:
     desc: Run tests
     cmds:
-      - go test -v -cover -race ./... {{ .CLI_ARGS }}
+      - go test -race -coverprofile=coverage.out -covermode=atomic -v ./... {{ .CLI_ARGS }}
+    silent: true
+
+  # internal (not directly called) tasks
+
+  install-go-tool:
+    internal: true
     silent: true
+    vars:
+      IS_TOOL_INSTALLED:
+        sh: which {{.GO_TOOL}} > /dev/null || echo "1"
+    cmds:
+      - test -z "{{.IS_TOOL_INSTALLED}}" || echo "Installing {{.GO_TOOL}}..."
+      - test -z "{{.IS_TOOL_INSTALLED}}" || go install {{.GO_TOOL_PATH}}
+      - test -n $(go env GOBIN) || go env -w GOBIN=$(go env GOPATH)/bin
+    requires:
+      vars: [GO_TOOL, GO_TOOL_PATH]
+
+  install-gofumpt:
+    desc: go install "gofumpt" and set GOBIN if not set
+    silent: true
+    cmds:
+      - task: install-go-tool
+        vars: { GO_TOOL: "gofumpt", GO_TOOL_PATH: "mvdan.cc/gofumpt@latest" }
+
+  install-golangci:
+    desc: go install "golangci-lint" and set GOBIN if not set
+    silent: true
+    cmds:
+      - task: install-go-tool
+        vars: { GO_TOOL: "golangci-lint", GO_TOOL_PATH: "github.com/golangci/golangci-lint/cmd/golangci-lint@v1.54.2" }
```

Comment:
can we add a comment to why the version is pinned vs using `@latest` ?

Reply:
I thought the golangci-lint page had a reason for pinning this, but I didn't see it. So I'm setting it to `@latest`

Reply:
🎉